sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java

```diff
@@ -0,0 +1,236 @@
+import com.azure.messaging.eventhubs.checkpointstore.blob.BlobCheckpointStore;
+import com.azure.messaging.eventhubs.models.ErrorContext;
+import com.azure.messaging.eventhubs.models.EventBatchContext;
+import com.azure.messaging.eventhubs.models.EventContext;
+import com.azure.messaging.eventhubs.models.EventPosition;
+import com.azure.messaging.eventhubs.stress.util.Constants;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Process events with options from application arguments. <br/>
+ * Required options: <br/>
+ *      --UPDATE_CHECKPOINT=YES/NO <br/>
+ *      --NEED_SEND_EVENT_HUB=YES/NO <br/>
+ */
+@Service("ProcessEventsWithOptions")
+public class ProcessEventsWithOptions extends EventHubsScenario {
+    private Logger logger = LoggerFactory.getLogger(ProcessEventsWithOptions.class);
+
+    @Override
+    public void run() {
+        String storageConnStr = options.get(Constants.STORAGE_CONNECTION_STRING);
+        String containerName = options.get(Constants.STORAGE_CONTAINER_NAME);
+        String eventHubConnStr = options.get(Constants.EVENT_HUBS_CONNECTION_STRING);
+        String eventHub = options.get(Constants.EVENT_HUB_NAME);
+        String consumerGroup = options.get(Constants.EVENT_HUB_CONSUMER_GROUP, EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME);
+
+        Objects.requireNonNull(options.get(Constants.UPDATE_CHECKPOINT), "UPDATE_CHECKPOINT argument required. value should be YES or NO");
+        Objects.requireNonNull(options.get(Constants.NEED_SEND_EVENT_HUB), "NEED_SEND_EVENT_HUB argument required. value should be YES or NO");
+        boolean updateCheckpoint = "YES".equalsIgnoreCase(options.get(Constants.UPDATE_CHECKPOINT));
+        boolean needSendEventHub = "YES".equalsIgnoreCase(options.get(Constants.NEED_SEND_EVENT_HUB));
+        String writeEventHub = options.get(Constants.SECOND_EVENT_HUB_NAME);
+        String writeEventHubConnStr = options.get(Constants.SECOND_EVENT_HUBS_CONNECTION_STRING);
+
+        final EventHubProducerClient producerClient = new EventHubClientBuilder()
+                .connectionString(writeEventHubConnStr, writeEventHub)
+                .buildProducerClient();
+
+
+        String batchSize = options.get(Constants.RECEIVE_BATCH_SIZE);
+        String batchTimeoutMs = options.get(Constants.RECEIVE_BATCH_TIMEOUT);
+
+        final int[] eventCounter = new int[64];
+        final ArrayList<ArrayList<EventData>> eventsFromPartitions = new ArrayList<ArrayList<EventData>>(64);
```

Comment:
Any reason for `ArrayList` instead of `List`, in type?

Reply:
Oh, I was copied from long running test directly. I'll change it part.